table/rows.go

```diff
@@ -0,0 +1,133 @@
+package table
+
+// Data is the interface that wraps the basic methods of a table model.
+type Data interface {
+	Row(row int) Row
+	Append(row Row)
```

Comment:
I think `Append` should remain optional. Not all data models can/will support that.